applications: serial_lte_modem: introduce POWER_PIN (#14112)
nrfconnect/sdk-nrf: applications/serial_lte_modem/doc/Generic_AT_commands.rst

* ``1`` - Enter Sleep.
  In this mode, both the SLM service and the LTE connection are terminated.

  The nRF91 Series SiP can be woken up using the :ref:`CONFIG_SLM_POWER_PIN <CONFIG_SLM_POWER_PIN>`.

* ``2`` - Enter Idle.

  In this mode, both the SLM service and the LTE connection are maintained.

>   The nRF91 Series SiP can be made to exit idle using the :ref:`CONFIG_SLM_POWER_PIN <CONFIG_SLM_POWER_PIN>`.

Reviewer: ```suggestion
  The nRF91 Series SiP can be made to exit idle using the pin specified with the :ref:`CONFIG_SLM_POWER_PIN <CONFIG_SLM_POWER_PIN>` Kconfig option`.
```

>   If the :ref:`CONFIG_SLM_INDICATE_PIN <CONFIG_SLM_INDICATE_PIN>` is defined, SLM toggles this GPIO when there is data for the MCU to read.

Reviewer: ```suggestion
  If the :ref:`CONFIG_SLM_INDICATE_PIN <CONFIG_SLM_INDICATE_PIN>` Kconfig option is defined, SLM toggles the specified pin when there is data for the MCU to read.
```

>   The MCU can in turn make SLM exit idle by toggling :ref:`CONFIG_SLM_POWER_PIN <CONFIG_SLM_POWER_PIN>`.

Reviewer: ```suggestion
  The MCU can in turn make SLM exit idle by toggling the pin specified with the :ref:`CONFIG_SLM_POWER_PIN <CONFIG_SLM_POWER_PIN>` Kconfig option.
```

.. note::

>    In this case the nRF91 Series SiP cannot be woken up using the :ref:`CONFIG_SLM_POWER_PIN <CONFIG_SLM_POWER_PIN>`.

Reviewer: ```suggestion
   In this case the nRF91 Series SiP cannot be woken up using the pin specified with the :ref:`CONFIG_SLM_POWER_PIN <CONFIG_SLM_POWER_PIN>` Kconfig option.
```
